tools/autograd/derivatives.yaml

```diff
@@ -911,29 +911,29 @@
   result: self_t / (1 + pow(2, other_p - self_p)) + other_t / (1 + pow(2, self_p - other_p))
 
 - name: xlogy.Tensor(Tensor self, Tensor other) -> Tensor
-  self: grad * at::xlogy((self != 0), other)
-  other: grad * at::where(other.isnan() | (self != 0), self / other, zeros_like(other))
-  result: self_t * at::xlogy((self_p != 0), other_p) + other_t * self_p / other_p
+  self: at::xlogy(grad, other)
+  other: grad * self / other
```

Comment:
This changes the value when other is nan or zero? Is that ok?

Reply:
When it's NaN we can return whatever we feel like. The previous value when `self` is zero was wrong and this fixes it.

Reply:
> When it's NaN we can return whatever we feel like.

That's not entirely true. We do try to preserve NaNs usually :p

Reply:
Oh, in that case this implementation is better than the previous one, which put zeros in the place of nans lol

Reply:
@alband in backward we don't (if forward function is not defined), we've had an issue about log recently, when someone was complaining of getting non-nan gradients for negative inputs.